File: packages/youtube-player/ios/Plugin/Classes/Events/PlaybackRateChangeEvent.swift
import Capacitor
import Foundation

@objc public class PlaybackRateChangeEvent: NSObject, Result {
    let id: String
    let rate: Double

    init(id: String, rate: Double) {
        self.id = id
        self.rate = rate
    }

    @objc public func toJSObject() -> AnyObject {
        var result = JSObject()
        result["id"] = id
        result["rate"] = rate
        return result as AnyObject
    }
}

File: packages/youtube-player/android/src/main/java/io/capawesome/capacitorjs/plugins/youtubeplayer/classes/events/PlaybackRateChangeEvent.java
package io.capawesome.capacitorjs.plugins.youtubeplayer.classes.events;

import androidx.annotation.NonNull;
import com.getcapacitor.JSObject;
import io.capawesome.capacitorjs.plugins.youtubeplayer.interfaces.Result;

public class PlaybackRateChangeEvent implements Result {

    @NonNull
    private final String id;

    private final double rate;

    public PlaybackRateChangeEvent(@NonNull String id, double rate) {
        this.id = id;
        this.rate = rate;
    }

    @Override
    @NonNull
    public JSObject toJSObject() {
        JSObject result = new JSObject();
        result.put("id", id);
        result.put("rate", rate);
        return result;
    }
}
